### PR TITLE
feat: add userguide

### PR DESCRIPTION
**Pull request by skyzh**
Signed-off-by: Alex Chi <iskyzh@gmail.com>

这个 PR 加入了 sjtuthesisuserguide.tex。这个文档的定位是一个最简的测试文件。开发者可以任意更改里面的内容，以测试 SJTUThesis 新添加的功能。

运行 `make build-dev` 即刻在 `sjtuthesis/doc` 目录下生成 pdf 文件。

### CODE REVIEW

File: sjtuthesis/build.lua [hunk 1]
```diff
@@ -4,13 +4,32 @@ module           = "sjtuthesis"
...
+
 bibtexexe        = "bibtex"
 
 unpackexe        = "xetex"
 unpackopts       = "--interaction=batchmode"
 unpackfiles      = {"sjtuthesis.dtx"}
+
+packtdszip       = true
+
+-- function type_example_latexmk()
```
Comment (LogCreative) [inline sjtuthesis/build.lua]: l3build provides two hooks for typesetting document demos:
- `typeset_demo_files()`
- `docinit_hook()`

The name of the hook might need to be changed.

**Review by LogCreative — COMMENTED** [review]
LGTM

File: sjtuthesis/build.lua (hunk 1 above)
Comment (LogCreative) [reply sjtuthesis/build.lua]: I ignored that you used `specialtypesetting` for that.😂

File: sjtuthesis/build.lua [hunk 2]
```diff
@@ -4,13 +4,32 @@ module           = "sjtuthesis"
...
 sourcefiledir    = "source"
 sourcefiles      = {"sjtuthesis.dtx"}
-installfiles     = {"*.sty"}
+installfiles     = {"*.cls","*.def","*.cfg","*.dtx","*.pdf"}
 
--- docfiledir       = "doc"
+docfiledir       = "doc"
 
 typesetexe       = "xelatex"
+typesetfiles     = {"sjtuthesisuserguide.tex","sjtuthesis.dtx"}
```
Comment (LogCreative) [inline sjtuthesis/build.lua:12]: When using `typesetfiles` for typesetting, you probably need to move the `sjtuthesisuserguide.tex` to `doc/` directory for compiling. I can't compile the userguide, at least on Windows.

Comment (skyzh) [reply sjtuthesis/build.lua:12]: In fact, I can compile this on both Arch Linux and macOS 🤣🤣

I'll investigate this later, let's hold this PR for now.

File: sjtuthesis/build.lua (hunk 1 above)
Comment (skyzh) [reply sjtuthesis/build.lua]: I planned to use that, but I found that `latexmk` doesn't work as expected. So I commented them out and use the default `xelatex` instead.

File: sjtuthesis/build.lua (hunk 2 above)
Comment (LogCreative) [reply sjtuthesis/build.lua:12]: WSL can not pass somehow, maybe l3build doesn't make a good job on Windows.🤔

Comment (LogCreative) [reply sjtuthesis/build.lua:12]: BTW, the artifact from GitHub Actions doesn't have the `userguide` as well. I moved the `.tex` to `doc/` folder, it is at least work on my machine.

File: sjtuthesis/build.lua (hunk 1 above)
Comment (LogCreative) [reply sjtuthesis/build.lua]: Yeah, `latexmk` is not working in `l3build`. Maybe they hate each other.

Comment (LogCreative) [reply sjtuthesis/build.lua]: A link for that: https://github.com/latex3/l3build/issues/113

Comment (LogCreative) [reply sjtuthesis/build.lua]: It may need to overwrite `typeset()` function to achieve that.🤪

**Review by LogCreative — APPROVED** [review]

**Review by AlexaraWu — APPROVED** [review]
非模板代码可以不用找我 Review  😂

**Pull request merged**